src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -517,7 +518,7 @@ internal static SecurityStatusPalErrorCode DoSslHandshake(SafeSslHandle context,
             sendCount = Crypto.BioCtrlPending(context.OutputBio!);
             if (sendCount > 0)
             {
-                sendBuf = new byte[sendCount];
+                sendBuf = ArrayPool<byte>.Shared.Rent(sendCount);
```

Comment:
There are lots of `Rent`s scattered over the PR and it seems difficult to figure out whether the arrays are actually getting returned. Maybe short comment mentioning when the buffer is returned would help?